flinkprocessor/src/main/java/io/pravega/example/videoprocessor/FaceRecognizerProcessor.java

```diff
@@ -0,0 +1,82 @@
+package io.pravega.example.videoprocessor;
+
+import io.pravega.example.common.Embedding;
+import io.pravega.example.common.Transaction;
+import io.pravega.example.common.VideoFrame;
+import io.pravega.example.tensorflow.*;
+import org.apache.flink.api.common.state.BroadcastState;
+import org.apache.flink.api.common.state.MapStateDescriptor;
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.streaming.api.functions.co.KeyedBroadcastProcessFunction;
+import org.apache.flink.util.Collector;
+import org.bytedeco.opencv.opencv_core.Mat;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+import static org.bytedeco.opencv.global.opencv_imgcodecs.IMREAD_UNCHANGED;
+import static org.bytedeco.opencv.global.opencv_imgcodecs.imdecode;
+
+/*
+ *   This class manages the embeddings database in state, and does facial recognition
+ * */
+public class FaceRecognizerProcessor
+        extends KeyedBroadcastProcessFunction<String, VideoFrame, Transaction, VideoFrame> {
+
+    private final Logger log = LoggerFactory.getLogger(FaceRecognizerProcessor.class);
+
+    private static EmbeddingsComparator embeddingsComp;
+
+    // broadcast state descriptor
+    MapStateDescriptor<String, Embedding> embeddingsDesc;
```

Comment:
Rename to `embeddingsDbDesc`.

Reply:
This class only supports one set of embeddings per person ID. It should support multiple. Embeddings should not be overwritten or deleted unless a delete transaction is received. embeddingsDbDesc should be a map from person ID to a List<Embedding>. You will need to update processBroadcastElement to reflect this.